src/pybind/mgr/nfs/export.py

```diff
@@ -880,3 +878,27 @@ def _apply_export(
     def _rados(self, cluster_id: str) -> NFSRados:
         """Return a new NFSRados object for the given cluster id."""
         return NFSRados(self.mgr.rados, cluster_id)
+
+    def get_export_count_with_same_fsal(self, cmount_path: str, cluster_id: str, fs_name: str) -> int:
+        exports = self.list_exports(cluster_id, detailed=True)
+        exports_count = 0
+        for export in exports:
+            if export['fsal']['name'] == 'CEPH' and export['fsal']['cmount_path'] == cmount_path and export['fsal']['fs_name'] == fs_name:
+                exports_count += 1
+        return exports_count
+
+
+def get_user_id(cluster_id: str, fs_name: str, cmount_path: str) -> str:
+    """
+    Generates a unique ID based on the input parameters using SHA-1.
+
+    :param cluster_id: String representing the cluster ID.
+    :param fs_name: String representing the file system name.
+    :param cmount_path: String representing the complicated mount path.
+    :return: A unique ID in the format 'cluster_id.fs_name.<hash>'.
+    """
+    input_string = f"{cluster_id}:{fs_name}:{cmount_path}"
+    hash_hex = hashlib.sha1(input_string.encode('utf-8')).hexdigest()
+    unique_id = f"{cluster_id}.{fs_name}.{hash_hex[:8]}"  # Use the first 8 characters of the hash
```

Comment:
same cluster, fs and cmount_path would in a combination would always generate same user_id, so only if this combination changes is the time when we have a new user. This would mean we have a key shared b/w all the exports using same cmount_path.

Reply:
> I do remember speaking about this in Ceph Days, but I ask again for the record and brevity - does this suffice to make the key distinct?

Yes it does suffice to make key distinct, also u can check screencast I attached [here](https://github.com/ceph/ceph/pull/54277#issuecomment-2262949962) it shows export creation & generated user_id in export info

Reply:
> > I do remember speaking about this in Ceph Days, but I ask again for the record and brevity - does this suffice to make the key distinct?
> 
> Yes it does suffice to make key distinct, also u can check screencast I attached [here](https://github.com/ceph/ceph/pull/54277#issuecomment-2262949962) it shows export creation & generated user_id in export info

that needs to be a test case

Reply:
there's unit test already and it does check user_id

Reply:
IIUC it's a mock test, i'm not sure if that suffices. By test case I mean in `qa/tasks/cephfs/test_nfs.py`

Reply:
So, truncating the sha to 8 chars suffices since there won't be lots and lots of exports. `git` uses 11 chars from the sha1 and that works well for repos with lots and lots of commits. So, I'd be fine with 8 byte truncate.

Reply:
yes, right! I think cephadm also uses only 6 chars for daemon_ids for a given service, so I  guess for exports 8 should suffice